File: sdk/dotnet/src/DependencyInjection/ServerSentEventScriptExtensions.fs
namespace StarFederation.Datastar.Scripts

open System.Runtime.CompilerServices
open StarFederation.Datastar.DependencyInjection

[<Extension>]
type ServerSentEventScriptExtensions() =

    [<Extension>]
    static member Redirect (sse:IDatastarServerSentEventService, url:string) =
        Redirect.redirect (sse.Handler, url)

    [<Extension>]
    static member BrowserConsoleAction (sse:IDatastarServerSentEventService, consoleAction:BrowserConsoleAction) =
        BrowserConsole.browserConsoleAction (sse.Handler, consoleAction)

File: sdk/dotnet/csharp/src/StarFederation.Datastar/Extensions/ServerSentEventScriptExtensions.cs
using StarFederation.Datastar.Enumerations;
using StarFederation.Datastar.Services;

namespace StarFederation.Datastar.Extensions;

/// <summary>
///     Extension methods for IDatastarServerSentEventService.
/// </summary>
public static class ServerSentEventScriptExtensions
{
    /// <summary>
    ///     Redirects the browser to the specified URL.
    /// </summary>
    /// <param name="service">The server-sent event service.</param>
    /// <param name="url">The URL to redirect to.</param>
    /// <returns>A task that represents the asynchronous redirect operation.</returns>
    public static Task RedirectAsync(this IDatastarServerSentEventService service, string url)
    {
        return RedirectExtensions.RedirectAsync(service, url);
    }

    /// <summary>
    ///     Executes a browser console action.
    /// </summary>
    /// <param name="service">The server-sent event service.</param>
    /// <param name="action">The action to execute.</param>
    /// <param name="message">The message for Log and Error actions.</param>
    /// <returns>A task that represents the asynchronous execution operation.</returns>
    public static Task BrowserConsoleActionAsync(this IDatastarServerSentEventService service, BrowserConsoleAction action, string? message = null)
    {
        return service.ExecuteConsoleActionAsync(action, message);
    }
}